### PR TITLE
Updates for 2.11

### PR DESCRIPTION
**Pull request by tpolecat**
Several issues ... I fixed them in arbitrary ways that we can discuss. I'll add inline comments.

:-1: tests may not run in 2.11 .. working on it

### CODE REVIEW

File: core/src/main/scala/package.scala [hunk 1]
```diff
@@ -48,6 +48,7 @@ package object funnel {
     def append(a: A, b: => A) = G.plus(a, b)
   }
 
-  implicit val dontUseTheDefaultStrategy: scalaz.concurrent.Strategy = null
-  implicit val theDefaultStrategyCausesProblems: scalaz.concurrent.Strategy = null
+  // This doesn't work in 2.11; these are at the highest priority within the package.
+  // implicit val dontUseTheDefaultStrategy: scalaz.concurrent.Strategy = null
+  // implicit val theDefaultStrategyCausesProblems: scalaz.concurrent.Strategy = null
 }
```
Comment (tpolecat) [inline core/src/main/scala/package.scala]: These bindings are at the highest priority in 2.11 which means it is not possible to use an implicit `Strategy` anywhere in package `funnel`. So they need to go somewhere else I think.

File: http/src/main/scala/MonitoringServer.scala [hunk 2]
```diff
@@ -91,11 +91,13 @@ class MonitoringServer(M: Monitoring, port: Int, keyTTL: Duration = 36.hours) {
...
+    import argonaut.EncodeJson
     val query = keyQuery(req.getRequestURI)
     val ks = M.keys.continuous.once.runLastOr(Set.empty).run.filter(x =>
       x.startsWith(prefix) && query(x))
-    val respBytes = JSON.prettyEncode(ks).getBytes
+    // rcn: this cannot be derived implicitly when Key's type arg is Any
+    val enc = EncodeJson.TraversableOnceEncodeJson[Key[Any],Set](JSON.EncodeKey[Any], implicitly)
+    val respBytes = JSON.prettyEncode(ks)(enc).getBytes
     flush(200, respBytes, req)
   }
```
Comment (tpolecat) [inline http/src/main/scala/MonitoringServer.scala]: I have no explanation for this. If it were `Key[String]` it would be ok, but the `Any` screws it up.

**Comment by timperrett** [comment]
@runarorama you should check this out also.

File: core/src/main/scala/package.scala (hunk 1 above)
Comment (runarorama) [inline core/src/main/scala/package.scala]: I think that's rather the point, that we don't want to use implicit Strategies in Funnel. No?

Comment (tpolecat) [inline core/src/main/scala/package.scala]: Ah ok, I see what you're doing. I think there are 4 cases where you're using the default `Strategy` that the 2.10 compiler just didn't catch.

```
[error] /Users/rnorris/Scala/funnel/core/src/main/scala/Monitoring.scala:278: ambiguous implicit values:
[error]  both value dontUseTheDefaultStrategy in package funnel of type => scalaz.concurrent.Strategy
[error]  and value theDefaultStrategyCausesProblems in package funnel of type => scalaz.concurrent.Strategy
[error]  match expected type scalaz.concurrent.Strategy
[error]     e(this).zip(alive.continuous).map(_._1).either(pts)
[error]                                                   ^
[error] /Users/rnorris/Scala/funnel/core/src/main/scala/Monitoring.scala:297: ambiguous implicit values:
[error]  both value dontUseTheDefaultStrategy in package funnel of type => scalaz.concurrent.Strategy
[error]  and value theDefaultStrategyCausesProblems in package funnel of type => scalaz.concurrent.Strategy
[error]  match expected type scalaz.concurrent.Strategy
[error]     e(this).zip(alive.continuous).map(_._1).either(pts)
[error]                                                   ^
[error] /Users/rnorris/Scala/funnel/core/src/main/scala/Monitoring.scala:393: ambiguous implicit values:
[error]  both value dontUseTheDefaultStrategy in package funnel of type => scalaz.concurrent.Strategy
[error]  and value theDefaultStrategyCausesProblems in package funnel of type => scalaz.concurrent.Strategy
[error]  match expected type scalaz.concurrent.Strategy
[error]   val costiveKeys: Signal[Set[Key[Any]]] = signalOf(Set())
[error]                                                    ^
[error] /Users/rnorris/Scala/funnel/core/src/main/scala/Monitoring.scala:510: ambiguous implicit values:
[error]  both value dontUseTheDefaultStrategy in package funnel of type => scalaz.concurrent.Strategy
[error]  and value theDefaultStrategyCausesProblems in package funnel of type => scalaz.concurrent.Strategy
[error]  match expected type scalaz.concurrent.Strategy
[error]      scalaz.stream.merge.mergeN(M.distinctKeys.filter(p).map(k => points(k)))
[error]                                ^
```

File: core/src/main/scala/Monitoring.scala [hunk 3]
```diff
@@ -60,10 +60,10 @@ trait Monitoring {
       buf: TBuffer[Option[I],O], costive: Boolean = false
     ): (Key[O], Task[I => Task[Unit]]) = {
       val k = keyMod(Key[O](name, units, description))
-      (k, topic(k, costive)(buf))
+      (k, topicWithKey(k, costive)(buf))
     }
 
-  def topic[I,O](
+  def topicWithKey[I,O](
```
Comment (tpolecat) [inline core/src/main/scala/Monitoring.scala]: So, in 2.11 you apparently can't have overloaded methods with default arguments, so I just renamed one of the overloads. You might prefer a different approach.

Comment (stew) [inline core/src/main/scala/Monitoring.scala]: :+1: fuck overloading

Comment (timperrett) [inline core/src/main/scala/Monitoring.scala]: +1... i cant even recall why we used overloading here.

File: core/src/main/scala/package.scala (hunk 1 above)
Comment (timperrett) [inline core/src/main/scala/package.scala]: @runarorama im leaning on the side of saying these should use `defaultPool` as their executor. WDYT?

**Comment by tpolecat** [comment]
Ok I added explicit strategies. All are `defaultPool` except the last one which uses the passed `EC`. No idea how sensible that is but at least the intent is correct.

**Comment by tpolecat** [comment]
Ok so I think this is actually ready for review. The explicit `Strategy` args are all `defaultPool` unless I found another strategy nearby. So it needs eyes that know what they're looking at. The integration tests **don't** work for me (see [log](https://gist.github.com/tpolecat/2747d318e1bc0a265aa2)) but they do work for Tim. Trying to figure out why.

File: core/src/main/scala/package.scala (hunk 1 above)
Comment (runarorama) [inline core/src/main/scala/package.scala]: Yes, `defaultPool` is the right thing here, I think.

**Comment by tpolecat** [comment]
Bah. I re-ran with an updated JDK and am still seeing [barrier failures](https://gist.github.com/tpolecat/a6750275379ba0e28cc8), so I'm not sure how to proceed. Happy to do more poking if you have ideas.

**Comment by timperrett** [comment]
@tpolecat something is very wrong with how the tests are running. i'm looking into it - might just exclude the integration tests for the time being as i wanted a better story for that anyway.

**Comment by tpolecat** [comment]
Ok, let me know if you need anything else from me on this PR. I can rebase/smash as needed.

**Pull request closed without merging**